internal/services/fluidrelay/fluid_relay_servers_resource.go

```diff
@@ -0,0 +1,399 @@
+	}
+	if s.IdentityType != "" {
+		res.Type = identity.Type(s.IdentityType)
+	}
+	if s.PrincipalID != "" {
+		res.PrincipalId = s.PrincipalID
+	}
+	if s.TenantID != "" {
+		res.TenantId = s.TenantID
+	}
+	res.IdentityIds = map[string]identity.UserAssignedIdentityDetails{}
+	for _, id := range s.UserAssignedIdentities {
+		res.IdentityIds[id.IdentityID] = identity.UserAssignedIdentityDetails{
+			ClientId:    utils.Ptr(id.ClientID),
+			PrincipalId: utils.Ptr(id.PrincipalID),
+		}
+	}
+	return res
+}
+
+func (s *ServerModel) GenEncryption() *fluidrelayservers.EncryptionProperties {
+	if len(s.Encryption) == 0 {
+		return nil
+	}
+	encryption := s.Encryption[0]
+	res := &fluidrelayservers.EncryptionProperties{
+		CustomerManagedKeyEncryption: &fluidrelayservers.CustomerManagedKeyEncryptionProperties{
+			KeyEncryptionKeyIdentity: &fluidrelayservers.CustomerManagedKeyEncryptionPropertiesKeyEncryptionKeyIdentity{
+				IdentityType:                   utils.Ptr(fluidrelayservers.CmkIdentityType(encryption.IdentityType)),
+				UserAssignedIdentityResourceId: utils.Ptr(encryption.IdentityResourceId),
+			},
+			KeyEncryptionKeyUrl: utils.Ptr(encryption.KeyEncryptionKeyUrl),
+		}}
+	return res
+}
+
+type Server struct{}
+
+var _ sdk.ResourceWithUpdate = (*Server)(nil)
+
+func (s Server) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			Description:  "The Fluid Relay server resource name",
+			ValidateFunc: validate.FluidRelayServerName,
+		},
+		"resource_group_name": commonschema.ResourceGroupName(),
+		"location":            commonschema.Location(),
+		"tags":                commonschema.Tags(),
+		"identity_type": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			Computed: true,
+			//Default:  "SystemAssigned",
+			ValidateFunc: validation.StringInSlice([]string{
+				"SystemAssigned",
+				"UserAssigned",
+				"SystemAssigned, UserAssigned",
+				"None",
+			}, false),
+		},
+		"user_assigned_identity": {
+			Type:        pluginsdk.TypeList,
+			Optional:    true,
+			Computed:    true,
+			ConfigMode:  pluginsdk.SchemaConfigModeBlock,
+			Description: "The list of user identities associated with the resource.",
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"identity_id": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						Computed:     true,
+						ValidateFunc: managedidentity.ValidateUserAssignedIdentitiesID,
+					},
+					"client_id": {
+						Type:        pluginsdk.TypeString,
+						Computed:    true,
+						Optional:    true,
+						Description: "The client id of user assigned identity.",
+					},
+					"principal_id": {
+						Type:        pluginsdk.TypeString,
+						Optional:    true,
+						Computed:    true,
+						Description: "The principal id of user assigned identity.",
+					},
+				},
+			},
+		},
+		"encryption": {
```

Comment:
from service team feedback, this property is not ready, should i delete it now?

> CMK is still under development and not ready yet. It will be ready in next few month.

Reply:
Sure